session.go

```diff
@@ -180,15 +155,14 @@ func GetSessionWithAccountIDAndPartition(c *Config) (*session.Session, string, s
 		}
 
 		accountID, partition, err := GetAccountIDAndPartition(iamClient, stsClient, credentialsProviderName)
-
-		if err == nil {
-			return sess, accountID, partition, nil
+		if err != nil {
```

Comment:
Happy path - return if there is an error. Otherwise continue happily down the path of normal execution.